files/create-ecr-repo/src/handler.py

```diff
@@ -28,27 +28,28 @@ def get_env_var(name, optional=False):
 
 
 # flake8: noqa: E501
-def lifecycle_policy():
-    policy = get_env_var(name="REPO_LIFECYCLE_POLICY", optional=True)
-    if policy:
+def get_valid_json_from_env(name, optional=True):
+    """
+    Function to get json content from environment variable,
+    validate them and return in marshal format.
+    """
+    payload = get_env_var(name=name, optional=optional)
+    if payload:
         try:
-            json.loads(policy)
+            json.loads(payload)
         except ValueError as err:
-            logger.error("Failed to parse REPO_LIFECYCLE_POLICY: %s", err)
+            logger.error("Failed to parse %s: %s", name, err)
+            if not optional:
+                sys.exit(1)
             return None
-    return policy
+    return payload
```

Comment:
Since `null` or `{}` values will make it all the way to the env variables, and the `get_valid_json_from_env("<variable>")` will return `'null'` or `{}` strings, which will evaluate to `True`, we'll be trying to set policies even if they are effectively empty.

How about we check JSON object inside the `get_valid_json_from_env()`, and return `None` if the parsed object is evaluated as `False`?

(also added `KeyError` exception handling, alongside with `ValueError`)

```python
def get_valid_json_from_env(name, optional=True):
    """
    Function to get json content from environment variable,
    validate them and return in marshal format.
    """
    payload = get_env_var(name=name, optional=optional)
    if payload:
        try:
            if json.loads(payload):
                return payload
        except (KeyError, ValueError) as err:
            logger.error("Failed to parse %s: %s", name, err)
            if not optional:
                sys.exit(1)
            return None
    return None
```

Reply:
I think that KeyError is unnecessary here, because `json.loads` [returns JSONDecodeError](https://docs.python.org/3/library/json.html#json.loads) and this is a [subclass of ValueError](https://docs.python.org/3/library/json.html#json.JSONDecodeError).

Also I checked what happened if pass `null` in terraform variable. Terraform will delete field from environment block after plan/apply.
 
But anyway I like your fix, it is cover more scenarios than mine. I pasted them as is and force pushed with signed.
Thank you! :pray: